### DISCUSSION TITLE
umi3 如何提升项目启动速度

### DISCUSSION THREAD

**Discussion by bluescurry** [umijs/umi#6067] [Q&A] [answered]
目前项目启动时间为 40 - 50s 之间，使用的是 umi 的最新包，想提升一下项目启动速度，组内小伙伴一直在吐槽这个启动速度太慢。。。想问问有哪些方法可以提升启动速度？
![image](https://user-images.githubusercontent.com/19237129/105620353-1a286800-5e37-11eb-86f4-d63fd43d5145.png)

**Comment by sorrycc** [comment, marked as answer]
1. umi ui 可以关了，不推荐使用了
2. 参考 https://umijs.org/guide/boost-compile-speed

**Reply by bluescurry** [reply]
1. 好的
2. 官方文档中写的编译提速策略我都做了，这个构建速度就是这些策略处理后的。想问一下为啥 webpack 的 dll umi3后面不用了

**Reply by AlisonZXQ** [reply]
> 1. 好的
> 2. 官方文档中写的编译提速策略我都做了，这个构建速度就是这些策略处理后的。想问一下为啥 webpack 的 dll umi3后面不用了

直接用splitchunks了

**Reply by banyudu** [reply]
splitChunks只是拆包，不会跳过，所以启动还是慢。它不能完全替代dll.

**Reply by kwzm** [reply]
+1 想使用 dll 把公共的 vendor 抽离出来并且不进行编译，splitChunks 只能做到抽离

**Comment by banyudu** [comment]
https://umijs.org/guide/boost-compile-speed 中的方法对于启动速度的提升并不明显。

是否可以考虑在dev模式中采用 snowpack 代替webpack？
